docs/frontend-architecture.md

```diff
@@ -54,3 +54,43 @@ then set its value in `next.config.js`.
 
 When the feature is rolled out to everyone, the flag can be removed by removing it from
 the mentioned `Record`. TypeScript will then point out everywhere it is used.
+
+## Add/modify mock data
+
+When you create a pull request, the frontend is deployed to Netlify using the code
+from that PR. However, the back-end is mocked out (using [Mock Service Worker](https://mswjs.io/)).
+You can also run the UI with a mocked back-end locally by running `npm run dev:mocked`
+in the `frontend` directory.
+
+The mock data is defined in `/frontend/src/apiMocks/mockData.ts`. For every API endpoint,
+different sets of data are defined for different user IDs, defined in the
+`mockIds` array. At the time of writing, the following mock users are available
+(IDs are inspired by [the nine states of design](https://medium.com/swlh/the-nine-states-of-design-5bfe9b3d6d85)):
+
+- `empty`: A user that just signed up for Relay, but has not created any aliases
+           yet, nor have they upgraded to Premium.
+- `onboarding`: A user that has just upgraded to Premium, but hasn't completed
+                the Premium onboarding flow yet.
+- `some`: A user that has an account that has seen some use: they've upgraded to
+          Premium, and have created some aliases.
+- `full`: A user that has utilised most of the features of Relay. They have
+          Premium, set up a custom domain, have both random and custom aliases,
+          and have experienced an email bounce.
+
+If you append `?mockId=<mockId>` (e.g. `?mockId=some`) to the URL, it will
+automatically log in as that mocked user. This is useful to quickly showcase a
+feature that's only visible in a particular state to e.g. a non-engineer.
+Alternatively, you can just click "Sign in", where you can choose between the
+different mock users.
+
+If you need to modify the mock data (e.g. because an API exposes some new data),
+you can do so by first updating the API type definitions in the applicable API
+endpoint's data fetching hook (in `/frontend/src/hooks/api/`). If you then look
+at `mockData.ts` in an editor that supports TypeScript (or run
+`npm run build:mocked`), you should then be guided through which objects need
+updating.
```

Comment:
praise: thanks! I was able to use this already in https://github.com/mozilla/fx-private-relay/pull/1713